src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -1736,12 +1737,12 @@ export class SearchResult extends Disposable {
 	private _createBaseFolderMatch(resource: URI | null, id: string, index: number, query: ITextQuery): FolderMatch {
 		let folderMatch: FolderMatch;
 		if (resource) {
-			folderMatch = this.instantiationService.createInstance(FolderMatchWorkspaceRoot, resource, id, index, query, this);
+			folderMatch = this._register(this.instantiationService.createInstance(FolderMatchWorkspaceRoot, resource, id, index, query, this));
 		} else {
-			folderMatch = this.instantiationService.createInstance(FolderMatchNoRoot, id, index, query, this);
+			folderMatch = this._register(this.instantiationService.createInstance(FolderMatchNoRoot, id, index, query, this));
 		}
-		const disposable = folderMatch.onChange((event) => this._onChange.fire(event));
-		folderMatch.onDispose(() => disposable.dispose());
+		const disposable = this._register(folderMatch.onChange((event) => this._onChange.fire(event)));
+		this._register(folderMatch.onDispose(() => disposable.dispose()));
 		return folderMatch;
```

Comment:
Two thoughts here:

1. Usually (but not always) when you create and return something for a consumer, it's the consumer's responsibility to dispose of that.
2. If many of these matches are created over the lifetime, you don't want to tie that to `this._register` since it will hold a reference to them until `this` is disposed

Reply:
I removed the `this._register` from the `const disposable = this._register(folderMatch.onChange(` since the consumer was already disposing it. I think that I had that from before I was handling folder disposal correctly. From my perspective, it still makes sense to have the folderMatch be registered and its `onDisposal` listener?